gax/test/test-application/src/index.ts

```diff
@@ -1017,43 +1081,55 @@ async function testServerStreamingRetrieswithRetryRequestOptionsResumptionStrate
   client.initialize();
 
   const request = createStreamingSequenceRequestFactory(
-    [Status.UNAVAILABLE, Status.DEADLINE_EXCEEDED, Status.OK],
-    [0.1, 0.1, 0.1],
-    [1, 2, 11],
+    [
+      Status.ALREADY_EXISTS, // Error code 6
+      Status.NOT_FOUND, // Error code 5
+      Status.DEADLINE_EXCEEDED, // Error code 4
+      Status.OK,
+    ],
+    [0.1, 0.1, 0.1, 0.1],
+    [0, 0, 0, 1],
     'This is testing the brand new and shiny StreamingSequence server 3'
   );
   const response = await client.createStreamingSequence(request);
-  const sequence = response[0];
+  await new Promise<void>((resolve, reject) => {
+    const sequence = response[0];
 
-  const attemptRequest =
-    new protos.google.showcase.v1beta1.AttemptStreamingSequenceRequest();
-  attemptRequest.name = sequence.name!;
+    const attemptRequest =
+      new protos.google.showcase.v1beta1.AttemptStreamingSequenceRequest();
+    attemptRequest.name = sequence.name!;
 
-  const attemptStream = client.attemptStreamingSequence(
-    attemptRequest,
-    settings
-  );
-  attemptStream.on('data', (response: {content: string}) => {
-    finalData.push(response.content);
-  });
-  attemptStream.on('error', () => {
-    throw new Error(
-      'testServerStreamingRetrieswithRetryRequestOptionsResumptionStrategy problem'
+    const attemptStream = client.attemptStreamingSequence(
+      attemptRequest,
+      settings
     );
-  });
-  attemptStream.on('end', () => {
-    assert.deepStrictEqual(
-      finalData.join(' '),
-      'This new and new and shiny StreamingSequence server 3'
-    );
-    attemptStream.end();
+    attemptStream.on('data', () => {
+      reject('The stream should not receive any data');
+    });
+    attemptStream.on('error', (error: GoogleError) => {
+      assert.strictEqual(error.code, 4);
+      assert.strictEqual(
+        error.message,
+        'Total timeout of API exceeded 2 milliseconds retrying error Error: 6 ALREADY_EXISTS: 6  before any response was received.'
```

Comment:
It's possible I need to update this to be less strict - I have also seen it error on the second error in the sequence

Reply:
I added a try catch here because I think it can be either the first or the second error code